core/src/main/java/org/infinispan/interceptors/impl/CacheWriterInterceptor.java

```diff
@@ -499,6 +499,9 @@ public int getNumberOfPersistedEntries() {
    }
 
    CompletionStage<Void> storeEntry(InvocationContext ctx, Object key, FlagAffectedCommand command, boolean incrementStats) {
+      if (persistenceManager.isReadOnly())
+         return CompletableFutures.completedNull();
+
```

Comment:
@wburns FYI, this optimisation to avoid unnecessary marshalling was accidentally removed in ISPN-9722. It resulted in the `HotRodConfigurationTest` encountered on CI, as it meant that when the CacheLoader was active, the JGroupAddress was being marshalled even though there was no store, resulting in a `NotSerializableException`.

Reply:
Yeah, not sure how I removed that :( Sorry about that.